### PR TITLE
Use `CoreDataStack` in `ThemeService`

### PR DESCRIPTION
**Pull request by crazytonyli**
Use `CoreDataStack` instead of `NSManagedObjectContext` in `ThemeService`. See the "Issue" and "Proposal" sections of https://github.com/wordpress-mobile/WordPress-iOS/pull/19893 for rational behind this change.

Even though there aren't many functions changed in this PR, but there are quite a few deletions, which might get in the way of code review. Like other PR, I'd recommend reviewing commit by commit, and turn on the "Hide Whitespace" option.

## Test instructions

As you can see in the changed files, there is only one screen affected, the "Themes" screen on "My Sites" tab. Verifying features on that screen still work should be enough, i.e. fetching and displaying the themes, activating a theme.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Services/ThemeService.m [hunk 1]
```diff
@@ -263,32 +162,42 @@ - (NSProgress *)getThemesForBlog:(Blog *)blog
...
+                                               if (![blogInContext.currentThemeId isEqualToString:deleteTheme.themeId]) {
+                                                   [context deleteObject:deleteTheme];
+                                               }
                                            }
                                        }
-                                   }
-
-                                   [[ContextManager sharedInstance] saveContext:self.managedObjectContext withCompletionBlock:^{
+                                       [context obtainPermanentIDsForObjects:themes error:nil];
+                                       themeObjectIDs = [themes wp_map:^id(Theme *obj) {
```
Comment (mokagio) [inline WordPress/Classes/Services/ThemeService.m:193]: I don't think I ever noticed these custom functions before. They're neat, but I wonder if we'd be betters served in using a third party library for this.

Or rather, I wonder if we'd be better served moving all the code that needs map, filter, reduce, etc. to the Swift layer where they're already part of the standard library and we don't need custom or third-party code to use them. 😁 

_One day..._

File: WordPress/Classes/ViewRelated/Themes/ThemeBrowserViewController.swift [hunk 2]
```diff
@@ -118,8 +118,9 @@ public protocol ThemePresenter: AnyObject {
         fetchRequest.fetchBatchSize = 20
         let sort = NSSortDescriptor(key: "order", ascending: true)
         fetchRequest.sortDescriptors = [sort]
-        let frc = NSFetchedResultsController(fetchRequest: fetchRequest,
-            managedObjectContext: self.themeService.managedObjectContext,
+        let frc = NSFetchedResultsController(
+            fetchRequest: fetchRequest,
+            managedObjectContext: self.themeService.coreDataStack.mainContext,
```
Comment (mokagio) [inline WordPress/Classes/ViewRelated/Themes/ThemeBrowserViewController.swift:123]: The SwiftLint part of me appreciates this change 🙇‍♂️

**Review by mokagio — APPROVED** [review]

File: WordPress/Classes/Services/ThemeService.m (hunk 1 above)
Comment (mokagio) [reply WordPress/Classes/Services/ThemeService.m:193]: FWIW: https://github.com/wordpress-mobile/WordPress-iOS/issues/20185

Comment (crazytonyli) [reply WordPress/Classes/Services/ThemeService.m:193]: 👍 for moving to Swift, which brings in whole lots of nice language features other than the array/collection operators.

**Pull request merged**